dp_consistency/src/correctoriginreport.py

```diff
@@ -0,0 +1,164 @@
+    except Exception as e:
+        print(f"An error occurred: {e}")
+
+    return (measurements, dfs)
+
+
+def getCorrectedReport(measurements):
+    report = Report(
+        {
+            ami: MetricReport(
+                total_campaign_reach_time_series=measurements[TOTAL_CAMPAIGN][
+                    AMI_FILTER
+                ],
+                reach_time_series_by_edp=[
+                    measurements[EDP_ONE][AMI_FILTER],
+                    measurements[EDP_TWO][AMI_FILTER],
+                ],
+            ),
+            mrc: MetricReport(
+                total_campaign_reach_time_series=measurements[TOTAL_CAMPAIGN][
+                    MRC_FILTER
+                ],
+                reach_time_series_by_edp=[
+                    measurements[EDP_ONE][MRC_FILTER],
+                    measurements[EDP_TWO][MRC_FILTER],
+                ],
+            ),
+        },
+        # AMI is a parent of MRC
+        metric_subsets_by_parent={ami: [mrc]},
+    )
+    return report.get_corrected_report()
+
+def correctSheetMetric(df, rows, func):
+    for period, (index, row) in enumerate(rows.iterrows()):
+        df.at[index, CUML_REACH_COL_NAME] = int(func(period).value)
+
+def correctSheet(df, ami_func, mrc_func):
+    ami_rows = df[df[FILTER_COL_NAME] == AMI_FILTER]
+    mrc_rows = df[df[FILTER_COL_NAME] == MRC_FILTER]
+    correctSheetMetric(df, ami_rows, ami_func)
+    correctSheetMetric(df, mrc_rows, mrc_func)
+    return df
+
+def buildCorrectedExcel(correctedReport, excel):
+    ami_metric_report = correctedReport.get_metric_report(ami)
+    mrc_metric_report = correctedReport.get_metric_report(mrc)
+    for edp in CUML_REACH_MAP:
+        amiFunc = (
+            ami_metric_report.get_total_reach_measurement
+            if (edp == TOTAL_CAMPAIGN)
+            else partial(
+                ami_metric_report.get_edp_measurement, CUML_REACH_MAP[edp]["ind"]
+            )
+        )
+        mrcFunc = (
+            mrc_metric_report.get_total_reach_measurement
+            if (edp == TOTAL_CAMPAIGN)
+            else partial(
+                mrc_metric_report.get_edp_measurement, CUML_REACH_MAP[edp]["ind"]
+            )
+        )
+
+        sheet_name = CUML_REACH_MAP[edp]['sheet']
+        excel[sheet_name] = correctSheet(
+            excel[sheet_name], amiFunc, mrcFunc
+        )
+    return excel
+
+
+def writeCorrectedExcel(path, corrected_excel):
+    with pd.ExcelWriter(path) as writer:
+        # Write each dataframe to a different sheet
+        for sheet_name in corrected_excel:
+            corrected_excel[sheet_name].to_excel(
+                writer, sheet_name=sheet_name, index=False
+            )
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Read an Excel file.")
+    parser.add_argument(
+        "--path_to_report", required=True, help="Path to the Excel file"
+    )
+    args = parser.parse_args()
+
+    (measurements, excel) = readExcel(args.path_to_report)
```

Comment:
Can we add a unit test for that include parsing and correcting a given excel file?